src/universal/modules/userDashboard/containers/UserSettings/UserSettingsContainer.js

```diff
@@ -76,7 +88,56 @@ export default class UserSettingsContainer extends Component {
       router.push(nextPage);
     }
     untouch('preferredName');
-  };
+  }
+
+  uploadPicture(pictureFile) {
+    return cashay.mutate('createUserPicturePutUrl', {
+      variables: {
+        userFilename: pictureFile.name
+      }
+    })
+    .then(({data, error}) => {
+      if (error) {
+        throw new Error(error._error); // eslint-disable-line no-underscore-dangle
+      }
+      return data.createUserPicturePutUrl.picturePutUrl;
+    })
+    .then(picturePutUrl => {
+      return fetch(picturePutUrl, {
```

Comment:
oh nevermind, the `picturePutUrl` is a signed hash... i was thinking plaintext url.